src/Application/Controller/Project/ReportController.php

```diff
@@ -0,0 +1,112 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ReportController extends AbstractController
+{
+    public function __construct(
+        private readonly UserManager $user_manager,
+        private readonly ProjectManager $project_manager,
+        private readonly TranslatorInterface $translator,
+        private readonly EventDispatcherInterface $event_dispatcher,
+        private readonly AuthorizationCheckerInterface $authorization_checker,
+        private readonly TokenStorageInterface $usage_tracking_token_storage,
+        private readonly EntityManagerInterface $entity_manager,
+    ) {
+    }
+
+    #[Route(path: '/api/reportProject/reportProject.json', name: 'api_report_program', defaults: ['_format' => 'json'], methods: ['POST', 'GET'])]
+    public function reportProject(Request $request): JsonResponse
+    {
+        $response = [];
+        $data = json_decode($request->getContent(), true);
+
+        if (!isset($data['program']) || !isset($data['category']) || !isset($data['note'])) {
+            $response['statusCode'] = 400; // Bad request
+            $response['answer'] = $this->translator->trans('errors.post-data', [], 'catroweb');
+            $response['preHeaderMessages'] = '';
+            $response['test'] = $data;
+
+            return new JsonResponse($response);
+        }
+
+        $category = strval($data['category']);
+        $note = strval($data['note']);
+        $projectId = strval($data['program']);
+        $project = $this->project_manager->find($projectId);
+        if (null == $project) {
+            $response['statusCode'] = 404; // Not found
+            $response['answer'] = $this->translator->trans('errors.project.invalid', [], 'catroweb');
+            $response['preHeaderMessages'] = '';
+
+            return new JsonResponse($response);
+        }
+
+        $report = new ProgramInappropriateReport();
+        $approved_project = $project->getApproved();
+        $featured_project = $this->project_manager->getFeaturedRepository()->isFeatured($project);
+        if ($approved_project || $featured_project) {
+            $response['answer'] = $this->translator->trans('success.report', [], 'catroweb');
+            $response['statusCode'] = Response::HTTP_OK;
+
+            return new JsonResponse($response);
+        }
+
+        $token = $request->headers->get('authorization');
+        if ($this->authorization_checker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
+            $user = $this->usage_tracking_token_storage->getToken()->getUser();
+            $report->setReportingUser($user);
+        } elseif (null !== $token) {
```

Comment:
this is already handled by our middleware, once the api call is protected in the security settings